website/blog/2024-05-22-whats-next.md

```diff
@@ -0,0 +1,75 @@
+---
+slug: whats-next
+title: Upcoming grants for MACI protocol improvements
+description: Dive into the next MACI protocol changes and improvements.
+authors:
+  name: ctrlc03
+  title: MACI dev
+  url: https://x.com/ctrlc03
+  image_url: https://avatars.githubusercontent.com/u/93448202?v=4
+tags: [voting, security, anonymity, roadmap, grants]
+---
+
+Minimal Anti Collusion Infrastructure ([MACI](https://github.com/privacy-scaling-explorations/maci)) is a public good that allows one to run secure, private, on-chain voting polls.
+
+Given MACI's open source nature, it's common for our core team to develop new features or to fix issues based on community feedback. However, it's been less common for external contributors to make significant changes to the core protocol.
+
+Well, this soon will be a reality thanks to a MACI improvement proposal sent by the [3327](https://3327.io/) team. 3327 is collective of 10+ people working on improving blockchain technologies, with a focus on research and engineering. Their development team will be working to implement significant changes to the MACI's codebase.
+
+This proposal can be divided into two parts:
+
+1. bring unconditional privacy to MACI's voters
+2. optimise inefficient merkle tree structure holding messages, by replacing it with a hash chain
+
+## 1) Provide unconditional voter privacy
+
+Currently with MACI, if a voter performs a [key change](/docs/core-concepts/key-change), the voter's new key would not be anonymous to the coordinator. The coordinator could collude with a bad actor to inform the latter of the key change, as the coordinator would have access to all decrypted messages.
+
+The key focus of this improvement is to allow users to be completely anonymous by removing the link between the original signup key and the key used for voting. How would this work? Well, users sign up to vote via the [MACI contract](/docs/developers-references/smart-contracts/MACI), and depending on the gatekeeper in use, they'd have to prove that they've passed the entry condition. Now, given knowledge of this key, they can signup with a new key to polls deployed by this same MACI contract.
```

Comment:
re-randomization

Reply:
https://github.com/0x3327/maci/blob/feat/elgamal/docs/elgamal-flow.md

Reply:
I'm not sure I follow you here, this work is a direct replacement of the ElGamal flow (but we should totally make it more clear that it is)

Reply:
sure if there's a sketch to the new flow that would be great, this is just the prior work on it to keep everything in one place. From the write up, looks like the main difference is that in PoC (IIRC) there was an explicit mixing phase between sign up and voting to anonymize your key, where as in the new flow the key change message can take place at any time.

Reply:
correct, though with this new flow the key change will not be a message per se, just a contract call with proof of inclusion and new key, to be stored in a poll specific tree. I believe 3327 named it anonymous poll joining which sounds like an appropriate name